development/contracts/ForkableBridge.sol

```diff
@@ -0,0 +1,155 @@
+pragma solidity ^0.8.17;
+
+import "@RealityETH/zkevm-contracts/contracts/PolygonZkEVMBridge.sol";
+import "@RealityETH/zkevm-contracts/contracts/lib/TokenWrapped.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/ClonesUpgradeable.sol";
+import "./interfaces/IForkableBridge.sol";
+
+contract ForkableBridge is PolygonZkEVMBridge, IForkableBridge {
+    address public forkmanager;
+    address public parentBridge;
+    address[] public children = new address[](2);
+
+    modifier onlyParent() {
+        require(msg.sender == parentBridge);
+        _;
+    }
+
+    modifier onlyForkManger() {
+        require(msg.sender == forkmanager);
+        _;
+    }
+
+    function initialize(
+        address _forkmanager,
+        address _parentBridge
+    ) external initializer {
+        forkmanager = _forkmanager;
+        parentBridge = _parentBridge;
+        // todo: overwrite the initialization once interfaces are correct.
+        // PolygonZkEVMBridge.initialize(_forkmanager, _parentBridge);
+    }
+
+    /**
+     * @notice Allows the forkmanager to create the new children
+     */
+    function createChildren() external onlyForkManger {
+        address forkableBridge = ClonesUpgradeable.clone(address(this));
+        // Todo: forkableBridge.initialize(super.networkId, super.globalExitRootManger.getFirstChild(),
+        // super.polygonZKEVMaddress.getFirstChild(), super.gasTokenAddress.getFirstChild(),super.isDeployedOnl2);
+        children[0] = forkableBridge;
+        forkableBridge = ClonesUpgradeable.clone(address(this));
+        // Todo: forkableBridge.initialize(super.networkId, super.globalExitRootManger.getFirstChild(),
+        // super.polygonZKEVMaddress.getFirstChild(), super.gasTokenAddress.getFirstChild(),super.isDeployedOnl2);
+        children[1] = forkableBridge;
+    }
+
+    /**
+     * @notice Anyone can use their tokens to split the bridged tokens into the two corresponding children tokens
+     * @param token token that should be split
+     * @param originNetwork origin network of the token to be split
+     * @param amount amount of tokens to be split
+     */
+    function splitTokenIntoChildTokens(
+        ERC20 token,
+        uint32 originNetwork,
+        uint256 amount
+    ) external {
+        require(children[0] != address(0), "Children not created yet");
+        require(children[1] != address(0), "Children not created yet");
+        require(
+            wrappedTokenToTokenInfo[address(token)].originNetwork != 0,
+            "Token not forkable"
+        );
+        TokenWrapped(address(token)).burn(msg.sender, amount);
+        bytes memory metadata = abi.encodePacked(
+            token.name(),
+            token.symbol(),
+            token.decimals()
+        );
+        ForkableBridge(children[0]).mintForkableToken(
```

Comment:
Use a call, or try catch in solidty, in order to prevent a bad token to sabotage the other operations